test/JDBC/expected/1_grant_schema.out

```diff
@@ -0,0 +1,2936 @@
+-- Need to terminate active session before cleaning up the login
+SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL)
+WHERE sys.suser_name(usesysid) = 'l1' AND backend_type = 'client backend' AND usesysid IS NOT NULL;
+go
+~~START~~
+bool
+t
+~~END~~
+
+
+-- Wait to sync with another session
+SELECT pg_sleep(1);
+go
+~~START~~
+void
+
+~~END~~
+
+
+-- tsql
+drop login l1;
+go
+
+-- tsql
+-- Create objects for tests in a user defined schema
+create schema s1;
+go
+create login l1 with password = '123';
+go
+create user u1 for login l1;
+go
+create login l2 with password = '123';
+go
+create user u2 for login l2;
+go
+create table s1.t1(a int);
+go
+create view s1.v1 as select 1;
+go
+create procedure s1.p1 as select 1;
+go
+create function s1.f1() returns int begin declare @a int; set @a = 1; return @a; end
+go
+create trigger s1.tr1 ON s1.t1 for insert as begin 
+select 'insert successful' end
+go
+create trigger s1.tr2 ON s1.t1 for insert as begin 
+grant select on s1.t1 to u1 end
+go
+create sequence s1.sq1 start with 1 increment by 1 ;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 2 [users have no default permission]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 (a) values (next value for s1.sq1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from s1.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec s1.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from s1.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql user=l2 password=123
+select current_user;
+go
+~~START~~
+varchar
+u2
+~~END~~
+
+select * from s1.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 (a) values (next value for s1.sq1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from s1.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec s1.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from s1.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql
+grant select on schema::s1 to u1;
+go
+grant insert on schema::s1 to u1;
+go
+grant delete on schema::s1 to u1;
+go
+grant update on schema::s1 to u1;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 3 [user u1 has select, insert, delete privilege on the schema]
```

Comment:
user u1 has select, insert, update and delete privilege on the schema

Reply:
Right! let me fix it.